src/pybind/mgr/balancer/module.py

```diff
@@ -1220,8 +1220,13 @@ def do_crush_compat(self, plan: MsPlan) -> Tuple[int, str]:
                                 'change balancer mode and retry might help'
 
     def get_compat_weight_set_weights(self, ms: MappingState):
-        if not CRUSHMap.have_default_choose_args(ms.crush_dump):
+        have_choose_args = CRUSHMap.have_default_choose_args(ms.crush_dump)
+        if have_choose_args:
+            # get number of buckets in choose_args
+            choose_args_len = len(CRUSHMap.get_default_choose_args(ms.crush_dump))
+        if not have_choose_args or choose_args_len != len(ms.crush_dump['buckets']):
```

Comment:
if `have_choose_args` is False, `choose_args_len` is not defined (or bound to any variable) here.